packages/flame/lib/src/layout/align_component.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:flame/src/anchor.dart';
+import 'package:flame/src/components/position_component.dart';
+import 'package:flame/src/effects/provider_interfaces.dart';
+import 'package:flutter/widgets.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+/// **AlignComponent** is a layout component that positions its child within
+/// itself using relative placement. It is similar to Flutter's [Align] widget.
+///
+/// The component requires a single [child], which will be the target of this
+/// component's alignment. Of course, other children can be added to this
+/// component too, but only the initial [child] will be aligned.
+///
+/// The [alignment] parameter describes where the child should be placed within
+/// the current component. For example, if the [alignment] is `Anchor.center`,
+/// then the child will be centered.
+///
+/// Normally, this component's size will match the size of its parent. However,
+/// if you provide properties [widthFactor] or [heightFactor], then the size of
+/// this component in that direction will be equal to the size of the child
+/// times the corresponding factor. For example, if you set [heightFactor] to
+/// 1 then the width of this component will be equal to the width of the parent,
+/// but the height will match the height of the child.
+///
+/// ```dart
+/// AlignComponent(
+///   child: TextComponent('hello'),
+///   alignment: Anchor.centerLeft,
+/// );
+/// ```
+///
+/// By default, the child's anchor is set equal to the [alignment] value. This
+/// achieves traditional alignment behavior: for example, the center of the
+/// child will be placed at the center of the current component, or bottom
+/// right corner of the child can be placed in the bottom right corner of the
+/// component. However, it is also possible to achieve more extravagant
+/// placement by giving the child a different anchor and setting
+/// [keepChildAnchor] to true. For example, if you set `alignment` to
```

Comment:
any reason you would _not_ want to keepChildAnchor?
why would you set an anchor to the child if you want to ignore it?

Reply:
Because when you receive a `child`, that child will have some non-null `anchor` value, and there is no way to know whether that anchor was set manually by the user or automatically by the framework.

Reply:
ah! got it :)